main.go

```diff
@@ -400,3 +410,28 @@ func runScorecardSettings(githubEventName, scorecardPolicyFile, scorecardResults
 	}
 	return &result, nil
 }
+
+func signScorecardResult(scorecardResultsFile string) error {
+	if err := os.Setenv("COSIGN_EXPERIMENTAL", "true"); err != nil {
+		return fmt.Errorf("error setting COSIGN_EXPERIMENTAL env var: %w", err)
+	}
+	ctx := context.Background()
+
+	// Prepare settings for SignBlobCmd.
+	keyOpts := sign.KeyOpts{
+		FulcioURL:    options.DefaultFulcioURL,     // Signing certificate provider.
+		RekorURL:     options.DefaultRekorURL,      // Transparency log.
+		OIDCIssuer:   options.DefaultOIDCIssuerURL, // OIDC provider to get ID token to auth for Fulcio.
+		OIDCClientID: "sigstore",
+	}
+	regOpts := options.RegistryOptions{} // Not necessary so we leave blank.
+
+	// This command will use the provided OIDCIssuer to authenticate into Fulcio, which will generate the
+	// signing certificate on the scorecard result. This attestation is then uploaded to the Rekor transparency log.
+	// The output bytes (signature) and certificate are discarded since verification can be done with just the payload.
+	if _, err := sign.SignBlobCmd(ctx, keyOpts, regOpts, scorecardResultsFile, true, "", "", time.Minute); err != nil {
```

Comment:
Signing this should be optional to the users. Some users might not want to do this, I would recommend we have a flag to enable/disable this setting.

Reply:
when users set the option `publish_results: true` in their workflow, we'll publish the results and sign them (if needed) for the badge. Otherwise this will be skipped. 

Does this resolve the concern?

Reply:
Yes, Thanks. We would  have to document in the action that publishing results also pushes it to rekor.